src/Config/GatewayAddresses.ts

```diff
@@ -0,0 +1,137 @@
+import MosaicConfig from '@openst/mosaic-chains/lib/src/Config/MosaicConfig';
+import GatewayConfig from '@openst/mosaic-chains/lib/src/Config/GatewayConfig';
+
+/**
+ * It represents contract addresses from mosaic config or token config.
+ */
+export default class GatewayAddresses {
+
+  public readonly valueTokenAddress: string;
+
+  public readonly baseTokenAddress: string;
+
+  public readonly stakePoolAddress: string;
+
+  public readonly originAnchorAddress: string;
+
+  public readonly auxiliaryAnchorAddress: string;
+
+  public readonly originGatewayAddress: string;
+
+  public readonly auxiliaryGatewayAddress: string;
+
+  public readonly redeemPoolAddress: string;
+
+  public readonly utilityTokenAddress: string;
+
+  public readonly originGatewayOrganizationAddress: string;
+
+  public readonly auxiliaryGatewayOrganizationAddress: string;
+
+  public readonly originAnchorOrganizationAddress: string;
+
+  public readonly auxiliaryAnchorOrganizationAddress: string;
+
+  /**
+   * Constructor.
+   * @param {string} valueTokenAddress
```

Comment:
No need to specify datatypes in comments. function arguments already has datatypes.
Also add comments in front of addresses.